src/NHapi.Base/Validation/IEncodingRule.cs

```diff
@@ -1,50 +1,48 @@
+
+  The Original Code is "EncodingRule.java".  Description:
+  "A validation rule that applies to encoded message strings.  "
+
+  The Initial Developer of the Original Code is University Health Network. Copyright (C)
+  2002.  All Rights Reserved.
 
-using System;
+  Contributor(s): ______________________________________.
+
+  Alternatively, the contents of this file may be used under the terms of the
+  GNU General Public License (the "GPL"), in which case the provisions of the GPL are
+  applicable instead of those above.  If you wish to allow use of your version of this
+  file only under the terms of the GPL and not to allow others to use your version
+  of this file under the MPL, indicate your decision by deleting  the provisions above
+  and replace  them with the notice and other provisions required by the GPL License.
+  If you do not delete the provisions above, a recipient may use your version of
+  this file under either the MPL or the GPL.
+*/
 
-namespace NHapi.Base.validation
+namespace NHapi.Base.Validation
 {
-   /// <summary> A validation rule that applies to encoded message strings.  Rules that 
-   /// apply to message content are not included in this category (they are MessageRules
-   /// and are evaluated against parsed Message object).  EncodingRules are intended 
-   /// for criteria that are specific to the encoded form of a message, e.g. "no empty 
-   /// tags in an XML message".  
-   /// </summary>
-   /// <author>  Bryan Tripp
-   /// </author>
-   public interface IEncodingRule : IRule
-	{
-		/// <summary> Tests the given encoded message text against the criteria 
-		/// defined by this rule class.  
-		/// </summary>
-		/// <returns> a list of exceptions indicating points at which the given 
-		/// message failed to validate (empty if validation succeeds; may 
-		/// not be a complete list as testing may be aborted after failure).  
-		/// </returns>
-		ValidationException[] test(String msg);
-	}
+    /// <summary> A validation rule that applies to encoded message strings.  Rules that
+    /// apply to message content are not included in this category (they are MessageRules
+    /// and are evaluated against parsed Message object).  EncodingRules are intended
+    /// for criteria that are specific to the encoded form of a message, e.g. "no empty
+    /// tags in an XML message".
+    /// </summary>
+    /// <author>  Bryan Tripp.
+    /// </author>
+    public interface IEncodingRule : IRule
+    {
+        /// <summary> Tests the given encoded message text against the criteria
+        /// defined by this rule class.
+        /// </summary>
+        /// <returns> a list of exceptions indicating points at which the given
+        /// message failed to validate (empty if validation succeeds; may
+        /// not be a complete list as testing may be aborted after failure).
+        /// </returns>
+        ValidationException[] test(string msg);

```

Comment:
Add Method with capital T and mark as obsolete?

Reply:
Was gonna ask you about these. The only reason I didn't was because this is a public interface, so adding methods is _technically_ a breaking change. What do you think? I think do it.

Reply:
I guess we would need to check what files are using the interface but I still reckon we should do it - if we didn't have to support net35 we could have potentially used the default interface implementation which just calls the original method 🤯 controversial or what 😁

Reply:
done